crates/btcio/src/reader/tx_indexer.rs

```diff
@@ -1,51 +1,73 @@
 use digest::Digest;
 use sha2::Sha256;
-use strata_l1tx::filter::visitor::OpsVisitor;
+use strata_l1tx::{filter::indexer::TxIndexer, messages::DaEntry};
+use strata_primitives::buf::Buf32;
 use strata_state::{
     batch::SignedBatchCheckpoint,
     tx::{DepositInfo, DepositRequestInfo, ProtocolOperation},
 };
 
-/// Ops visitor for rollup client. This is intended to get Da commitment as well as process/store da
-/// blob.
+/// Ops indexer for rollup client. Collects extra info like da blobs and deposit requests
 #[derive(Clone, Debug)]
-pub struct ClientOpsVisitor {
+pub struct ClientTxIndexer {
     ops: Vec<ProtocolOperation>,
-    // TODO: Add l1 manager to store da to db
+    deposit_requests: Vec<DepositRequestInfo>,
+    da_entries: Vec<DaEntry>,
 }
 
-impl ClientOpsVisitor {
+impl ClientTxIndexer {
     pub fn new() -> Self {
-        Self { ops: Vec::new() }
+        Self {
+            ops: Vec::new(),
+            deposit_requests: Vec::new(),
+            da_entries: Vec::new(),
+        }
     }
-}
 
-impl OpsVisitor for ClientOpsVisitor {
-    fn collect(self) -> Vec<ProtocolOperation> {
-        self.ops
+    fn ops(&self) -> &[ProtocolOperation] {
+        &self.ops
     }
+}
 
+impl TxIndexer for ClientTxIndexer {
     fn visit_da<'a>(&mut self, chunks: impl Iterator<Item = &'a [u8]>) {
         let mut hasher = Sha256::new();
+        let mut blob = Vec::new();
+
         for chunk in chunks {
             hasher.update(chunk);
+            blob.extend_from_slice(chunk);
         }
         let hash: [u8; 32] = hasher.finalize().into();
-        // TODO: store da in db
-        self.ops.push(ProtocolOperation::DaCommitment(hash.into()));
+        let commitment: Buf32 = hash.into();
```

Comment:
Abstract this out to use the DA commitment type directly.  Avoid duplication for sensitive things like this, even though there's not much space to mess up.